airflow/providers/google/cloud/hooks/bigquery_reservation.py

```diff
@@ -0,0 +1,123 @@
+#
```

Comment:
I wonder if it's better to name it `hooks/bigquery_biengine.py` ?

Reply:
+1 for `hooks/bigquery_biengine.py`

Reply:
I have thought about it but it uses the bigquery-reservation API. So, I would prefer keep this name. Moreover, this could be extended with other bigquery-reservation api routes like create and delete bigquery slot reservation (so using the same client). 

PS: To be honest, It should be my next PR (create and delete BigQuery slots reservation operators) but I wish to validate all the process with BI Engine first.

Reply:
However, I am open, if you prefer this naming, I will do it.

Reply:
In general we prefer to keep things organized under service and not under API.
you can see we have bigquery.py, bigtable.py etc..
It makes it easier to find what you need when everything is under the same service.

Reply:
Thanks for your detailed answer, I will change.